File: 1470-shuffle-the-array/1470-shuffle-the-array.cpp
class Solution {
public:
    vector<int> shuffle(vector<int>& nums, int n) {
        int firstNum, secondNum;
        for (int i = n ; i < 2 * n; ++i){
            secondNum = nums[i] << 10;
            nums[i - n] |= secondNum;
        }
        
        int allOnes = int(pow(2, 10)) - 1;
        
        for (int i = n - 1; i >= 0; --i){
            firstNum = nums[i] & allOnes;
            secondNum = nums[i] >> 10;
            
            nums[2 * i] = firstNum;
            nums[2 * i + 1] = secondNum;
        }
        
        return nums;
    }
};

File: 1470-shuffle-the-array/1470-shuffle-the-array.py
class Solution:
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        x_sec, y_sec = len(nums) // 2, len(nums)
        ans = []
        for i in range(x_sec):
            x_n = nums[i]
            y_n = nums[x_sec + i]
            ans.extend([x_n, y_n])
            
        return ans